code/modules/mob/living/carbon/alien/diona/diona_powers.dm

```diff
@@ -53,17 +159,132 @@
 		src.verbs -= /mob/living/carbon/alien/diona/proc/split
 		return
 
-	src.loc << "You feel a pang of loss as [src] splits away from your biomass."
-	src << "You wiggle out of the depths of [src.loc]'s biomass and plop to the ground."
+	var/r = alert(src,"Splitting will remove you from your gestalt and deposit you on the ground, allowing you to go it alone. If you had any stored biomass before you joined the gestalt, you will not get it back. Are you sure you wish to split?", "Confirm Split", "Time to leaf", "I'll stick around")
+	if (r != "Time to leaf")
+		return
+
+
+	src.loc << "\red You feel a pang of loss as [src] splits away from your biomass."
+	src << "<span class='notice'>You wiggle out of the depths of [src.loc]'s biomass and plop to the ground.</span>"
 
-	var/mob/living/M = src.loc
+	if (gestalt.is_diona() == DIONA_NYMPH)
+		gestalt.nutrition -= NYMPH_ABSORB_NUTRITION//Preventing an exploit with repeatedly absorbing and splitting
 
+	split_languages(gestalt)
 	src.loc = get_turf(src)
-	src.verbs -= /mob/living/carbon/alien/diona/proc/split
-	src.verbs += /mob/living/carbon/alien/diona/proc/merge
+	stat = CONSCIOUS
+	gestalt = null
+	update_verbs()
+
+
+
+//Draws a sizeable blood sample from a victim to read their DNA and learn languages
+/mob/living/carbon/alien/diona/proc/sample()
+	set category = "Abilities"
+	set name = "Sample DNA"
+	set desc = "Learn languages by draining some blood from a nearby lifeform. You will partially learn any language it knows."
+
+	//For fun factor, we'll allow the nymph to choose nonvalid targets
+	var/list/choices = list()
+	for(var/mob/living/L in view(1,src))
 
-	if(istype(M))
-		for(var/atom/A in M.contents)
-			if(istype(A,/mob/living/simple_animal/borer) || istype(A,/obj/item/weapon/holder))
+		if((!(src.Adjacent(L)) || L == src)) continue
+		choices.Add(L)
+
+	if (!choices.len)
+		src << "\red There are no life forms nearby to sample!"
```

Comment:
When writing new code, start preferring `<span class='danger'>` and so on, instead of `\red` and other macros. For future reference, this one.